### PR TITLE
Supporting disguised mid-circuit measurements

### PR DESCRIPTION
**Pull request by obliviateandsurrender**
A mid-circuit measurement may appear to be terminal since no other operation happens on the wire, but in reality, it might be used for some upcoming controlled workflow elsewhere in the circuit. This adds a checking condition to catch such scenarios and treat those terminal measurements as mid-circuit measurements.

### CODE REVIEW

**Comment by trbromley** [comment]
Thanks @obliviateandsurrender for this quick fix!

One thing I'm confused about, do we need to conceptually treat terminal and mid-circuit measurements differently at all?

**Comment by obliviateandsurrender** [comment]
@trbromley iirc the main inspiration to treat them differently was the fact that when users provide their own `measurements` then we would replace the terminal measurements in the circuit while still retaining the mid-circuit ones.

**Review by Mandrenkov — APPROVED** [review]
Looks good to me! 🎉 

It could be worth simplifying the test to make its intent more clear but the more realistic coverage is nice too.

**Review by lillian542 — APPROVED** [review]
Looks good to me 🚀

File: pennylane_qiskit/converter.py [hunk 1]
```diff
@@ -481,12 +481,24 @@ def _function(*args, params: dict = None, wires: list = None, **kwargs):
...
                 meas_terminal = True
-                for next_op, next_qargs, __ in qc.data[idx + 1 :]:
-                    # Check if the subsequent whether next_op is measurement interfering
+                for next_op, next_qargs, next_cargs in qc.data[idx + 1 :]:
+                    # Check if the subsequent conditional is measurement needing
+                    if isinstance(next_op, ControlFlowOp):
+                        if set(next_cargs).intersection(op_cregs):
+                            meas_terminal = False
+                            break
+                    elif next_op.condition:  # For legacy c_if
```
[the remark this answers is not in the document]
Comment (obliviateandsurrender) [reply pennylane_qiskit/converter.py:494]: No, it is not.

**Pull request merged**